x-pack/plugins/secops/server/lib/events/query.dsl.ts

```diff
@@ -60,34 +66,48 @@ export const buildQuery = (options: EventsRequestOptions) => {
       }
     : {};
 
-  return {
+  const queryMust = options.fields.includes('kpiEventType')
+    ? [{ match_all: {} }, { exists: { field: 'event.type' } }]
+    : [{ match_all: {} }];
+
+  let sort: SortRequest = [];
+  if (sortFieldId) {
+    const field: string = sortFieldId === 'timestamp' ? '@timestamp' : sortFieldId;
+    const dir: SortRequestDirection = direction === 'descending' ? 'desc' : 'asc';
+    sort = [...sort, { [field]: dir }, { [options.sourceConfiguration.fields.tiebreaker]: dir }];
```

Comment:
I might be missing something, but it looks like `sort` is not modified between where it's declared as an empty array with a `let` on line `73`:

`let sort: SortRequest = [];`

and where it is spread over near the start of line `77`:

`sort = [...sort, { [field]: dir }, { [options.sourceConfiguration.fields.tiebreaker]: dir }];`

If that observation is correct:
a) do we need the `...sort,`
b) can we replace `let` with a `const` with something like:

```
const getSortField = ({ sortFieldId, direction }: { sortFieldId: string, direction: string }) => {
    const field: string = sortFieldId === 'timestamp' ? '@timestamp' : sortFieldId;
    const dir: SortRequestDirection = direction === 'descending' ? 'desc' : 'asc';

    return [{ [field]: dir }];
}

const sort: SortRequest = (sortFieldId) ? [
  ...getSortField({ sortFieldId, direction }),
  { [options.sourceConfiguration.fields.tiebreaker]: dir }
] : [];
```

?